Data/HashMap/Base.hs

```diff
@@ -1883,6 +1890,33 @@ fromListWith :: (Eq k, Hashable k) => (v -> v -> v) -> [(k, v)] -> HashMap k v
 fromListWith f = L.foldl' (\ m (k, v) -> unsafeInsertWith f k v m) empty
 {-# INLINE fromListWith #-}
 
+-- | /O(n*log n)/ Construct a map from a list of elements.  Uses
+-- the provided function to merge duplicate entries.
+--
+-- === Examples
+--
+-- Given a list of key-value pairs where the keys are of different flavours, e.g:
+--
+-- > data Key = Mul | Add
```

Comment:
I think the result of the example below depends on the Hashable instance, so you should probably include it here.

Reply:
Yes indeed, the result depends on the `Hashable` instance. But I left it out because I felt that it was a distraction and didn't really add anything important. There are only two instances here, one which will give you the order that I provided in the example, and the other which will give you the reverse order. Since the order isn't really the point of the example I'd rather leave out the instance.